### PR TITLE
Use `/proc/[pid]/map_files`

### PR DESCRIPTION
**Pull request by danielocfb**
With this change we start using proc map_files entries when attempting to access files backing a certain memory region. These entries are valid while a process runs, even if the backing file has been deleted (or replaced) on the file system. They are also correct in case of file system name space usage, whereas the proc maps path may only have meaning inside or outside the namespace. As such, they are the correct piece of information to use.
Unfortunately, reading files through the "symbolic links" below map_files requires as additional capability, CAP_SYS_ADMIN (see proc(5)). As such, we now start running our unit tests as root. We can consider falling back to other means, such as file access through /proc/[pid]/root/ should this pose an unacceptable restriction, but for now we error on the side of simplicity.

We remove the two existing load_symbolfilecfg_* tests. These arguably are hacks relying on specific paths to be reported and those paths change with this patch.

### CODE REVIEW

File: src/maps.rs [hunk 1]
```diff
@@ -108,17 +162,26 @@ fn parse_maps_line<'line>(line: &'line str, pid: Pid) -> Result<MapsEntry> {
...
         .map(|(_inode, line)| line.trim())
         .unwrap_or("");
-    let path = if path_str.ends_with(" (deleted)") {
-        PathBuf::from(format!("/proc/{pid}/map_files/{address_str}"))
-    } else {
-        PathBuf::from(path_str)
+
+    let path_name = match path_str.as_bytes() {
+        [] => None,
+        [b'[', .., b']'] => Some(PathName::Component(path_str.to_string())),
```
Comment (anakryiko) [inline src/maps.rs:168]: it's not always `[something]` for non-file entries, so best not to assume this format. I think just checking `/` at the beginning should be fine to distinguish files on disk from anything special. And later on we can explicitly check for `[vdso]` (as an example), when we need to process them specially.

Here are few examples from real process on my machine:

```
7ff8d9eab000-7ff8d9ecc000 rw-s 00000000 00:0e 2057                       anon_inode:[perf_event]
7ff8d9ecc000-7ff8d9eed000 rw-s 00000000 00:0e 2057                       anon_inode:[perf_event]
7ff8d9f2d000-7ff8d9f2e000 r--s 00000000 00:0e 2057                       anon_inode:bpf-map
7ff8d9f6f000-7ff8d9f70000 r--s 00000000 00:0e 2057                       anon_inode:bpf-map
```

File: src/maps.rs [hunk 2]
```diff
@@ -182,31 +245,30 @@ pub(crate) fn parse(pid: Pid) -> Result<impl Iterator<Item = Result<MapsEntry>>>
...
+        range,
+        mode,
+        offset,
+        path_name,
+    } = entry;
 
     // Only entries that are executable and readable (r-x-) are of relevance.
-    if (entry.mode & 0b1010) != 0b1010 {
-        return false
+    if (mode & 0b1010) != 0b1010 {
```
Comment (anakryiko) [inline src/maps.rs:259]: is there a reason to dictate that mapping has to be only readable? I don't know if that actually matters in practice, but the less assumptions we make, the better. So maybe let's just check that executable bit is set?

Comment (danielocfb) [reply src/maps.rs:259]: Yeah, I thought about that, too. I can adjust.

File: src/maps.rs (hunk 1 above)
Comment (danielocfb) [reply src/maps.rs]: I adjusted the code accordingly. Thanks!

**Review by anakryiko — APPROVED** [review]
Looks great!

**Pull request merged**